lib/src/html/html_generator.dart

```diff
@@ -79,9 +79,20 @@ class HtmlGenerator extends Generator {
       if (!enabled) {
         throw new StateError('`write` was called after `generate` completed.');
       }
-      // If you see this assert, we're probably being called to build non-canonical
-      // docs somehow.  Check data.self.isCanonical and callers for bugs.
-      assert(allowOverwrite || !writtenFiles.contains(filePath));
+
+      assert(
+          !writtenFiles
+              .map((String file) => file.toLowerCase())
+              .contains(filePath.toLowerCase()),
+          'Filename case metamer found. Output would not be correct on a '
```

Comment:
I don't see where writtenFiles is being populated with lowercase names.

also nit: once it is, this case is actually the union of the previous case and the metamer case.

Reply:
Oh, right.  That's likely to be really slow, as we'll convert the entire set every time, but yes it will work.

Reply:
Yeah, I think I'll just make two lists (one actual case, and one lowercase) to make it more efficient.